File: backend/src/main/java/Models/Club.java
package Models;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Club {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;


    private String name;

    private String logo;
    
    private String banner;
    
    private String description;
    
    private int creationYear;

    @ManyToOne
    private User responsible;

    @OneToOne
    private Content content;
    

}

File: application/lib/Models/Club.dart
import 'package:enicar_social_media/Models/User.dart';

class Club {
  int? id;
  String? name;
  String? logo;
  String? banner;
  String? description;
  int? creationYear;
  User? responsible;
  String? content;

  Club(
      {this.id,
      this.name,
      this.logo,
      this.banner,
      this.description,
      this.creationYear,
      this.responsible,
      this.content});

  Club.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    logo = json['logo'];
    banner = json['banner'];
    description = json['description'];
    creationYear = json['creationYear'];
    responsible =
        json['responsible'] != null ? User.fromJson(json['responsible']) : null;
    content = json['content'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'logo': logo,
      'banner': banner,
      'description': description,
      'creationYear': creationYear,
      'responsible': responsible!.toJson(),
      'content': content,
    };
  }
}
